src/main/java/edu/cornell/kfs/sys/service/impl/CUMarshalServiceImpl.java

```diff
@@ -141,6 +152,19 @@ public <T> T unmarshalStream(InputStream inputStream, Class<T> clazz)
         Unmarshaller unmarshaller = createUnmarshaller(clazz, Optional.empty());
         return unmarshalStream(inputStream, unmarshaller);
     }
+    
+    @Override
+    public <T> T unmarshalStreamIgnoreDtd(InputStream inputStream, Class<T> clazz)
+            throws JAXBException, IOException, XMLStreamException {
+        JAXBContext jc = JAXBContext.newInstance(clazz);
+
+        XMLInputFactory xif = XMLInputFactory.newFactory();
+        xif.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+        XMLStreamReader xsr = xif.createXMLStreamReader(inputStream);
```

Comment:
You may want to close the reader when you're done with it. The `CuXMLStreamUtils` class has a convenience method for closing it quietly.

Also, if the caller is supposed to be responsible for closing the input stream, then please wrap the stream argument in a `CloseShieldInputStream` before creating the reader. There should be an example in one of this class's other methods.